### PR TITLE
Use RetryableSphereClientBuilder for client creation

### PR DESCRIPTION
**Pull request by salander85**

### CODE REVIEW

File: service/src/test/java/com/commercetools/util/SphereClientConfigurationUtilTest.java [hunk 1]
```diff
@@ -36,74 +22,20 @@
...
         when(tenantConfig.getSphereClientConfig())
-                .thenReturn(SphereClientConfig.of("test-key", "test-client-id", "test-client-secret"));
+            .thenReturn(SphereClientConfig.of("test-key", "test-client-id", "test-client-secret"));
     }
 
     @Test
-    public void executeCreateClient_isBlocking() {
-        SphereClient sphereClient = SphereClientConfigurationUtil.createClient(tenantConfig.getSphereClientConfig());
+    public void createBlockingClient_WithConfig_returnsBlockingSphereClient() {
+        SphereClient sphereClient =
```
Comment (leungkinghin) [inline service/src/test/java/com/commercetools/util/SphereClientConfigurationUtilTest.java:30]: Add final keywords

**Comment by leungkinghin** [comment]
Great!
Btw, this PR cannot be merged to master branch at this moment. It is suggested to make sure functional tests run successfully in travis after Payone investigated the redirect link issue.

**Review by leungkinghin — APPROVED** [review]

File: service/src/test/java/com/commercetools/util/SphereClientConfigurationUtilTest.java (hunk 1 above)
Comment (salander85) [reply service/src/test/java/com/commercetools/util/SphereClientConfigurationUtilTest.java]: f068650

File: service/src/main/java/com/commercetools/util/SphereClientConfigurationUtil.java [hunk 2]
```diff
@@ -1,99 +1,55 @@
...
 
-import static io.sphere.sdk.http.HttpStatusCode.BAD_GATEWAY_502;
-import static io.sphere.sdk.http.HttpStatusCode.GATEWAY_TIMEOUT_504;
-import static io.sphere.sdk.http.HttpStatusCode.SERVICE_UNAVAILABLE_503;
-
 public final class SphereClientConfigurationUtil {
     private static final long DEFAULT_TIMEOUT = 10;
     private static final TimeUnit DEFAULT_TIMEOUT_TIME_UNIT = TimeUnit.SECONDS;
-    protected static final int RETRIES_LIMIT = 5;
-    private static final int MAX_PARALLEL_REQUESTS = 30;
```
Comment (salander85) [inline service/src/main/java/com/commercetools/util/SphereClientConfigurationUtil.java:28]: @leungkinghin Default value for maximum parallel requests is now 20; do you think we should increase to 30 or use the defaults?

Comment (leungkinghin) [reply service/src/main/java/com/commercetools/util/SphereClientConfigurationUtil.java:28]: I would prefer to keep it as 30, it is default value used in payone-integration before the change.

Comment (salander85) [reply service/src/main/java/com/commercetools/util/SphereClientConfigurationUtil.java:28]: 4e6901d

**Pull request merged**